duckduckgo/macos-browser
# Improve Chromium Logins import

Task/Issue URL: https://app.asana.com/0/1177771139624306/1202077207271006/f
Tech Design URL:
CC: @samsymons 

**Description**:
Users signed into Google Account in Chrome/Chromium can use their
cloud-stored passwords (that are also stored locally in a separate
file called "Login Data For Account"). For every newly saved login
they can also decide whether to store it in the cloud (and in
"Login Data For Account") or locally (in "Login Data").

This change takes these two files into consideration when importing
passwords from Chrome. In case of duplicates (decided based on the
site URL + username combination) the entry with a newer timestamp
is imported (judging based on `date_password_modified`).


**Steps to test this PR**:
1. Remove all passwords from Chrome and enable using passwords stored in a Google Account
1. Save two logins in Google Account: site A and site B
1. Disable using passwords stored in a Google Account
1. Save two logins locally: site A and site C
1. Re-enable using passwords from a Google Account (this will re-import cloud-stored passwords)
1. Import passwords into DuckDuckGo
1. Observe that 3 passwords were imported

**Testing checklist**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
